Sources/VSM/StateContainer/StateObserving.swift

```diff
@@ -8,70 +8,70 @@
 import Combine
 import Foundation
 
-/// Provides functions for observing an action for potentially new states
+/// Provides functions for observing VSM actions to render new states on the view.
 public protocol StateObserving<State> {
     associatedtype State
     
-    /// Observes the state publisher emitted as a result of invoking some action
+    /// Renders the states emitted by the publisher on the view.
+    /// - Parameter statePublisher: The view state publisher to be observed for rendering the current view state
     func observe(_ statePublisher: AnyPublisher<State, Never>)
-
-    /// Observes the state emitted as a result of invoking some synchronous action
+    
+    /// Renders the next state on the view.
+    /// - Parameter nextState: The next view state to render
     func observe(_ nextState: State)
     
-    /// Observes the state emitted as a result of invoking some asynchronous action
+    /// Asynchronously renders the next state on the view.
+    /// - Parameter nextState: The next view state to render
     func observeAsync(_ nextState: @escaping () async -> State)
     
-    /// Observes the states emitted as a result of invoking some asynchronous action that returns an asynchronous sequence
+    /// Asynchronously renders the sequence of states on the view.
+    /// - Parameter stateSequence: The sequence of states to render
     func observeAsync<StateSequence: AsyncSequence>(_ stateSequence: @escaping () async -> StateSequence) where StateSequence.Element == State
     
     // MARK: - Debounce
     
-    /// Debounces the action calls by `dueTime`, then observes the `State` publisher emitted as a result of invoking the action.
-    /// Prevents actions from being excessively called when bound to noisy UI events.
-    /// Action calls are grouped by the provided `identifier`.
+    /// Renders the states emitted by the publisher on the view.
+    /// Calls to this function are debounced to prevent excessive execution from noisy events.
     /// - Parameters:
     ///   - statePublisher: The action to be debounced before invoking
     ///   - dueTime: The amount of time required to pass before invoking the most recent action
-    ///   - identifier: The identifier for grouping actions for debouncing
+    ///   - identifier: (optional) The identifier for grouping actions for debouncing
     func observe(
         _ statePublisher: @escaping @autoclosure () ->  AnyPublisher<State, Never>,
         debounced dueTime: DispatchQueue.SchedulerTimeType.Stride,
         identifier: AnyHashable
```

Comment:
The doc says _optional_. Should the actual parameter be optional with a nil default? (Question applies as well below several more times.)

Reply:
Normally, yes. However, because the alternative overload uses `file: StaticString = #fileID, line: UInt = #line` parameters to get the identity from the caller, I had to use a bit of a different approach to hide those default args from those using the protocol.